fhir-server/src/test/java/com/ibm/fhir/server/test/FHIRRestHelperTest.java

```diff
@@ -0,0 +1,987 @@
+        Bundle.Entry.Request bundleEntryRequest = Bundle.Entry.Request.builder().method(HTTPVerb.POST).url(Uri.of("Patient")).build();
+        Bundle.Entry bundleEntry = Bundle.Entry.builder().fullUrl(Uri.of("urn:1")).resource(patient).request(bundleEntryRequest).build();
+        
+        Procedure procedure = Procedure.builder().status(ProcedureStatus.COMPLETED).subject(Reference.builder().reference(com.ibm.fhir.model.type.String.of("urn:1")).build()).build();
+        Bundle.Entry.Request bundleEntryRequest2 = Bundle.Entry.Request.builder().method(HTTPVerb.POST).url(Uri.of("Procedure")).build();
+        Bundle.Entry bundleEntry2 = Bundle.Entry.builder().resource(procedure).request(bundleEntryRequest2).build();
+        
+        Bundle requestBundle = Bundle.builder().id("bundle1").type(BundleType.TRANSACTION).entry(bundleEntry2, bundleEntry).build();
+
+        // Process bundle
+        FHIRRequestContext.get().setOriginalRequestUri("test");
+        FHIRRequestContext.get().setReturnPreference(HTTPReturnPreference.REPRESENTATION);
+        Bundle responseBundle = helper.doBundle(requestBundle, null);
+        
+        // Validate results
+        assertNotNull(responseBundle);
+        assertEquals(2, responseBundle.getEntry().size());
+        for (Bundle.Entry entry : responseBundle.getEntry()) {
+            Bundle.Entry.Response response = entry.getResponse();
+            if (response.getLocation().getValue().startsWith("Patient")) {
+                assertEquals("Patient/generated-0/_history/1", response.getLocation().getValue());
+                assertEquals(Integer.toString(Response.Status.CREATED.getStatusCode()), response.getStatus().getValue());
+            } else if (response.getLocation().getValue().startsWith("Procedure")) {
+                assertEquals("Procedure/generated-1/_history/1", response.getLocation().getValue());
+                assertEquals(Integer.toString(Response.Status.CREATED.getStatusCode()), response.getStatus().getValue());
+                Procedure returnedProcedure = (Procedure) entry.getResource();
+                assertEquals("Patient/generated-0", returnedProcedure.getSubject().getReference().getValue());
+            } else {
+                fail();
+            }
+        }
+    }
+
+    /**
+     * Test transaction bundle post with circular local reference dependency.
+     * Procedure has local reference to Encounter and Encounter has local reference to Procedure.
+     */
+    @Test
+    public void testTransactionBundlePostWithCircularDependency() throws Exception {
+        FHIRPersistence persistence = new MockPersistenceImpl();
+        FHIRRestHelper helper = new FHIRRestHelper(persistence);
+        
+        Encounter encounter = Encounter.builder().status(EncounterStatus.FINISHED).clazz(Coding.builder().code(Code.of("AMB")).build())
+                .reasonReference(Reference.builder().reference(com.ibm.fhir.model.type.String.of("urn:2")).build()).build();
```

Comment:
this style is totally fine, but often times we'll prefer a style with more whitespace like this:
```suggestion
        Encounter encounter = Encounter.builder()
                .status(EncounterStatus.FINISHED)
                .clazz(Coding.builder()
                    .code(Code.of("AMB"))
                    .build())
                .reasonReference(Reference.builder()
                    .reference(com.ibm.fhir.model.type.String.of("urn:2"))
                    .build())
                .build();
```

Reply:
additionally, we'll often statically import the `com.ibm.fhir.model.type.String.string` factory method so that we don't need to fully quality `com.ibm.fhir.model.type.String` for each usage.

Reply:
these style comments apply to many of the tests in this class...not planning to repeat it for each instance

Reply:
I updated the testcases to use the style suggested, as well as importing `com.ibm.fhir.model.type.String.string`